chainstate/src/detail/tokens/mod.rs

```diff
@@ -31,14 +31,14 @@ mod check_utils;
 pub use check_utils::is_rfc3986_valid_symbol;
 use check_utils::{check_nft_description, check_nft_name, check_token_ticker, is_uri_valid};
 
-pub fn check_tokens_transfer_data(
+pub fn check_positive_amount(
```

Comment:
You can create NonZeroAmount instead of thism just like NonZeroU32.

Reply:
removed this function and added in-place check. I cannot change the type as it goes to the db